opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -2184,12 +2184,18 @@ namespace Opm {
             const double dt = this->simulator_.timeStepSize();
             // Calculate common THP for subsea manifold well group (item 3 of NODEPROP set to YES)
             computeWellGroupThp(dt, deferred_logger);
```

Comment:
this should be in the sub iteration loop.

Reply:
Are you sure about that? I think it should be outside. This is potentially expensive to evaluate. I have done some testing with auto choke and this seems to speed up the calculations without affecting the results significantly. I may change my mind as this with further testing and as the auto choke support advances. Any comments @plgbrts?

Reply:
In theory, probably, since other node pressure change (upstream nodes above the auto-choke node) might cause the manifold group can not produce the target rate, but for that case, we do not need to go through the expensive process to adjust.  

For now, we can go with your testing.

Reply:
I tested with https://github.com/OPM/opm-simulators/pull/5754 and needed to do some changes to get similar results. I added a check if the group_well_thp was updated. With this change the results looks similar, but the run time is reduced by a factor 3. 

![image](https://github.com/user-attachments/assets/4210dbaf-7012-4395-af04-596d417b8d24)